tests/test_asset.py

```diff
@@ -5,91 +5,149 @@
+                    "data.json",
+                ]
+                if request.param == "unpack"
+                else ["output.tgz"]
+            )
+            return sample_files
+
+    @pytest.fixture()
+    def expected_stac_files(self, requests_mock: req_mock.Mocker) -> str:
+        sample_stac = pathlib.Path(__file__).resolve().parent / "mock_data/multipolygon.geojson"
+        with open(sample_stac, "rb") as src_file:
+            out_file = src_file.read()
+            requests_mock.get(
+                url=constants.STAC_ASSET_URL,
+                content=out_file,
+            )
+            return "bsg-104-20230522-044750-90756881_ortho.tiff"
+
+    def test_should_delegate_repr_to_info(self):
+        assert repr(asset.Asset(asset_info=self.asset_info)) == repr(self.asset_info)
+
+    def test_should_initialize(self, requests_mock: req_mock.Mocker):
+        url = host.endpoint(f"/v2/assets/{constants.ASSET_ID}/metadata")
+        requests_mock.get(url=url, json=self.asset_info)
+        assert asset.Asset(asset_id=constants.ASSET_ID).info == self.asset_info
+
+    def test_should_initialize_with_info_provided(self):
+        assert asset.Asset(asset_info=self.asset_info).info == self.asset_info
+
+    @pytest.mark.parametrize(
+        "asset_id, asset_info, expected_error",
+        [
+            (
+                None,
+                None,
+                "Either asset_id or asset_info should be provided in the constructor.",
+            ),
+            (
+                constants.ASSET_ID,
+                {"id": constants.ASSET_ID},
+                "asset_id and asset_info cannot be provided simultaneously.",
+            ),
+        ],
+        ids=[
+            "Both asset_id and asset_info provided",
+            "Neither asset_id and asset_info provided",
+        ],
+    )
+    def test_fails_initialize_with_concurrent_or_empty_info_id(
+        self, asset_id: Optional[str], asset_info: Optional[dict], expected_error: str
+    ):
+        with pytest.raises(ValueError, match=expected_error):
+            asset.Asset(asset_id=asset_id, asset_info=asset_info)
+
+    def test_should_download_expected_files(
+        self, requests_mock: req_mock.Mocker, output_directory: Optional[str], expected_files: List[str]
+    ):
+        url = host.endpoint(f"/v2/assets/{constants.ASSET_ID}/metadata")
+        unpacking = len(expected_files) > 1
+        requests_mock.get(url=url, json=self.asset_info)
+        asset_obj = asset.Asset(asset_id=constants.ASSET_ID)
+        requests_mock.post(
+            url=f"{constants.API_HOST}/v2/assets/{constants.ASSET_ID}/download-url",
+            json={"url": constants.DOWNLOAD_URL},
+        )
+        downloaded_files = asset_obj.download(output_directory, unpacking=unpacking)
+        assert len(downloaded_files) == len(expected_files)
+        assert all([pathlib.Path(p).exists() and pathlib.Path(p).name in expected_files for p in downloaded_files])
```

Comment:
1. assert set(downloaded_files) == set(expected_files)
2. assert all(pathlib.Path(p).exists() for name in downloaded_files)